shm/Store.cpp

```diff
@@ -1,33 +1,152 @@
 #include "Store.hpp"
 
 #include <algorithm>
+#include <random>
 
-Store::Store(std::shared_ptr<Time>& time)
-    : time_(time) {
-    time_->addObserver(this);
+#include "Alcohol.hpp"
+#include "Fruit.hpp"
+#include "Item.hpp"
+
+Store::Store(std::shared_ptr<Time>& time) {
+    time_ = time;
+    time->addObserver(this);
+    market_ = makeStock(generateFruits(), generateAlcos(), generateItems());
+}
+
+size_t Store::generateRandom(int min, int max) const {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distirb(min, max);
+
+    return static_cast<size_t>(distirb(gen));
+}
+
+Cargos Store::generateFruits() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Fruit fruit(fruitNames[generateRandom(0, 5)], generateRandom(1, 20), generateRandom(10, 30), generateRandom(1, 10));
+        if(std::none_of(begin(result), end(result),[&fruit](const auto& ptr){ return ptr->getName() == fruit.getName();})) {
+            result.emplace_back(std::make_shared<Fruit>(fruit));
+            i++;
+        }
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
+
+Cargos Store::generateAlcos() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Alcohol alco(alcoNames[generateRandom(0, 5)], generateRandom(1, 5), generateRandom(30, 100), generateRandom(40, 96));
+        if(std::none_of(begin(result), end(result),[&alco](const auto& ptr){ return ptr->getName() == alco.getName();})) {
+            result.emplace_back(std::make_shared<Alcohol>(alco));
+            i++;
+        }
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
+
+Cargos Store::generateItems() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Item item(itemNames[generateRandom(0, 5)], generateRandom(1, 10), generateRandom(30, 100), Rarity::rare);
+        if(std::none_of(begin(result), end(result),[&item](const auto& ptr){ return ptr->getName() == item.getName();})) {
+            result.emplace_back(std::make_shared<Item>(item));
+            i++;
+        }
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
```

Comment:
I've seen you comment that you're not sure how to randomize rarity values. 

I've done something like that:
```cpp
void Store::generateItems() {
    for (auto name : itemNames_) {
        std::random_device rd;
        std::mt19937 gen(rd());
        std::uniform_int_distribution<> randAmount(0, 20);
        std::uniform_int_distribution<> randBasePrice(50, 100);
        std::uniform_int_distribution<> randRarity(0, rarityAmount - 1);
        cargo_.emplace_back(std::make_shared<Item>(name, randAmount(gen), randBasePrice(gen), possibleRarities[randRarity(gen)]));
    }
}
```

and `rarityAmount` and `possibleRarities` are defined in Item.hpp file as:
```cpp
constexpr size_t rarityAmount{4};
constexpr Rarity possibleRarities[rarityAmount]{Rarity::common, Rarity::rare, Rarity::epic, Rarity::legendary};
```

So it's basically taking a randomized-index element array.

Reply:
I also had an idea to change vector of item names to map<name, rarity>, so we can make unique items and avoid situations like same items having diffrent rarity in various stores. Also in this was we can make legendary items very rare and common items more common